### ISSUE TITLE
[Bug]: Build broken related to Material library after adding compose-auth-ui

### ISSUE THREAD

**Issue by hieuwu** [supabase-community/supabase-kt#684]
### General Info

- [X] I checked for similar bug report
- [X] I am using the latest version
- [X] I checked the [troubleshooting](https://github.com/supabase-community/supabase-kt/blob/master/TROUBLESHOOTING.md) page for similar problems

### Version(s)

2.4.0

### Kotlin Target(s) and their respective versions

JVM 17, Android 13, Gradle 8.2.2

### What happened? (include your code)

Here is my dependencies with Supabase
```gradle
implementation(platform("io.github.jan-tennert.supabase:bom:2.4.0"))
implementation("io.github.jan-tennert.supabase:postgrest-kt")
implementation("io.github.jan-tennert.supabase:gotrue-kt")
implementation("io.github.jan-tennert.supabase:storage-kt")
```
However when I add 
`implementation("io.github.jan-tennert.supabase:compose-auth-ui")`

The code breaks at imports from material3 while I can still use composable from `compose-auth-ui` normally. Android Studio suggests that I should add material library as below:
<img width="650" alt="Screenshot 2024-08-03 at 21 24 00" src="https://github.com/user-attachments/assets/c75f35b3-f39d-4df0-a340-6f399932d8b5">

### Steps To Reproduce (optional)

1. Use
```kotliin
import androidx.compose.material3.DismissState
import androidx.compose.material3.rememberDismissState
```
2. Add `implementation("io.github.jan-tennert.supabase:compose-auth-ui")` to gradle then build
3. Build Debug

**Comment by jan-tennert** [comment]
This is probably because `compose-auth-ui` uses a newer Material3 library version. `remberDismissState` and `DismissState` have been renamed in a `1.2.X` version. [See the related changelog](https://developer.android.com/jetpack/androidx/releases/compose-material3?hl=de#1.2.0-beta02)

**Comment by hieuwu** [comment]
Thank Jan that's the reason. The issue is resolved when I bump material 3 version to `1.2.1`
